tests/integration/p2p.go

```diff
@@ -398,12 +413,16 @@ func setupReplicatorWaitSync(
 		ready <- struct{}{}
 		for waitIndex := 0; waitIndex < len(sourceToTargetEvents); waitIndex++ {
 			for i := 0; i < targetToSourceEvents[waitIndex]; i++ {
-				err := sourceNode.WaitForPushLogByPeerEvent(targetPeerInfo.ID)
-				require.NoError(s.t, err)
+				msg, ok := <-sourceSub.Message()
```

Comment:
todo: Please document why you are doing this, it is unclear reading the code (same goes for the other similar block).  

todo: Please also document what `sourceSub` and `targetSub` do in the `setupReplicatorWaitSync` definition.

Reply:
moved to a helper func and added docs